tests/src/fully_connected_test.c

```diff
@@ -145,34 +145,39 @@ void esp_nn_fully_connected_per_channel_s8_test()
     int32_t input_offset = 0;
     int32_t filter_offset = 0;
     int32_t out_offset = 7;
-
-    // both these arrays should be sized equal to `out_channels`, in the test cases we go up to max 16 channels hence this size.
-    // for tests with smaller num of channels, we just use the first `out_channels` indexes.
-    int32_t out_mult[max_out_ch];
-    int32_t out_shift[max_out_ch];
+    int32_t* out_mult;
+    int32_t* out_shift;
 
     printf("\n######## Running %s ##########\n", __FUNCTION__);
     for (int itr = 0;  itr < 15; itr++) {
         switch (itr) {
         case 0:
+            out_mult = malloc(out_channels * sizeof(int32_t));
+            out_shift = malloc(out_channels * sizeof(int32_t));
```

Comment:
Please follow the pattern from [here](https://github.com/espressif/esp-nn/blob/master/tests/src/convolution_test.c) for allocation, sanity check, and if the pointer is no NULL before free.